var/spack/repos/builtin/packages/lfortran/package.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Lfortran(Package):
+    """LFortran"""
+
+    homepage = "https://lfortran.org"
+
+    git      = "https://gitlab.com/lfortran/lfortran.git"
+
+    version('master', branch="master")
+    version('0.15.0', tag="v0.15.0")
+
+    depends_on('python@3:', type='build')
+    depends_on('cmake', type='build')
+    depends_on('llvm@14.0.4', type='build')
+    depends_on('zlib@1.2.12', type='build')
+    depends_on('re2c@2.2', type='build')
+    depends_on('bison@3.4.2', type='build')
+    depends_on('binutils@2.38', type='build', when='platform=linux')
+
+    maintainers = ['certik']
+
+    def install(self, spec, prefix):
+        open("version", "w").write(str(self.version))
+
+        python = which("python")
+        python("grammar/asdl_cpp.py")
+        python("grammar/asdl_cpp.py", "src/libasr/ASR.asdl", "src/libasr/asr.h")
+
+        re2c = which("re2c")
+        bison = which("bison")
+        with working_dir("src/lfortran/parser"):
+            re2c("-W", "-b", "tokenizer.re", "-o", "tokenizer.cpp")
+            re2c("-W", "-b", "preprocessor.re", "-o", "preprocessor.cpp")
+            bison("-Wall", "-d", "-r", "all", "parser.yy")
```

Comment:
Run those steps in the preparing / patching stage.

Reply:
Done!